Detectors/Upgrades/ALICE3/EC0/simulation/src/Detector.cxx

```diff
@@ -0,0 +1,557 @@
+
+#include "SimulationDataFormat/Stack.h"
+#include "SimulationDataFormat/TrackReference.h"
+
+// FairRoot includes
+#include "FairDetector.h"    // for FairDetector
+#include "FairLogger.h"      // for LOG, LOG_IF
+#include "FairRootManager.h" // for FairRootManager
+#include "FairRun.h"         // for FairRun
+#include "FairRuntimeDb.h"   // for FairRuntimeDb
+#include "FairVolume.h"      // for FairVolume
+#include "FairRootManager.h"
+
+#include "TGeoManager.h"     // for TGeoManager, gGeoManager
+#include "TGeoTube.h"        // for TGeoTube
+#include "TGeoPcon.h"        // for TGeoPcon
+#include "TGeoVolume.h"      // for TGeoVolume, TGeoVolumeAssembly
+#include "TString.h"         // for TString, operator+
+#include "TVirtualMC.h"      // for gMC, TVirtualMC
+#include "TVirtualMCStack.h" // for TVirtualMCStack
+
+#include <cstdio> // for NULL, snprintf
+
+class FairModule;
+
+class TGeoMedium;
+
+class TParticle;
+
+using std::cout;
+using std::endl;
+
+using namespace o2::ec0;
+using o2::itsmft::Hit;
+
+//_________________________________________________________________________________________________
+Detector::Detector()
+  : o2::base::DetImpl<Detector>("EC0", kTRUE),
+    mTrackData(),
+    mHits(o2::utils::createSimVector<o2::itsmft::Hit>())
+{
+}
+
+//_________________________________________________________________________________________________
+void Detector::buildBasicEC0(int nLayers, Float_t z_first, Float_t z_length, Float_t etaIn, Float_t etaOut, Float_t Layerx2X0)
+{
+  // Build a basic parametrized EC0 detector with nLayers equally spaced between z_first and z_first+z_length
+  // Covering pseudo rapidity [etaIn,etaOut]. Passive silicon thinkness computed to match layer x/X0
+
+  mNumberOfLayers = nLayers;
+  Float_t sensorThickness = 30.e-4;
+  mLayerName.resize(mNumberOfLayers);
+  mLayerID.resize(mNumberOfLayers);
+
+  for (int layerNumber = 0; layerNumber < mNumberOfLayers; layerNumber++) {
+    std::string layerName = GeometryTGeo::getEC0LayerPattern() + std::to_string(layerNumber);
+    mLayerName[layerNumber] = layerName;
+
+    // Adds evenly spaced layers
+    Float_t layerZ = z_first + (layerNumber * z_length / (mNumberOfLayers - 1)) * std::copysign(1, z_first);
+    Float_t rIn = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaIn))));
+    Float_t rOut = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaOut))));
+
+    auto& thisLayer = mLayers.emplace_back(layerNumber, layerName, layerZ, rIn, rOut, sensorThickness, Layerx2X0);
+  }
+}
+
+//_________________________________________________________________________________________________
+void Detector::buildEC0V1()
+{
+  //Build EC0 detector according to
+  //https://indico.cern.ch/event/992488/contributions/4174473/attachments/2168881/3661331/tracker_parameters_werner_jan_11_2021.pdf
+
+  mNumberOfLayers = 10;
+  Float_t sensorThickness = 30.e-4;
+  Float_t layersx2X0 = 1.e-2;
+  std::vector<std::array<Float_t, 5>> layersConfig{
```

Comment:
Could it be possible to add also the backward part of the detector here?
Configuration should be the symmetric mirroring of the FWD part.

Reply:
Done